partner_second_lastname/models/res_partner.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# Copyright 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+# Copyright 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+
+from odoo import api, fields, models
+from odoo.addons.partner_firstname import exceptions
+
+
+class ResPartner(models.Model):
+    """Adds a second last name."""
+    _inherit = "res.partner"
+
+    lastname2 = fields.Char("Second last name", oldname="lastname_second")
+
+    @api.model
+    def _get_computed_name(self, lastname, firstname, lastname2=None):
+        """Compute the name combined with the second lastname too.
+
+        We have 2 lastnames, so lastnames and firstname will be separated by a
+        comma.
+        """
+        order = self._get_names_order()
+        names = list()
+        if order == 'first_last':
+            if firstname:
+                names.append(firstname)
+            if lastname:
+                names.append(lastname)
+            if lastname2:
+                names.append(lastname2)
+        else:
+            if lastname:
+                names.append(lastname)
+            if lastname2:
+                names.append(lastname2)
+            if names and firstname and order == 'last_first_comma':
+                names[-1] = names[-1] + ","
+            if firstname:
+                names.append(firstname)
+        return u" ".join(names)
+
+    @api.depends("firstname", "lastname", "lastname2")
+    def _compute_name(self):
+        """Write :attr:`~.name` according to splitted data."""
+        for partner in self:
+            partner.name = self._get_computed_name(
+                partner.lastname, partner.firstname, partner.lastname2,
+            )
+
+    @api.one
```

Comment:
I know, but I don't want to change this logic right now (not too much time for this) :wink: